### PR TITLE
Add @types/uuid helper functions

### PR DESCRIPTION
**Pull request by Crazyglue**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by ctavan** [comment]
This is a duplicate of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46395

**Review by RyanCavanaugh — APPROVED** [review]

File: types/uuid/interfaces.d.ts [hunk 1]
```diff
@@ -48,3 +48,12 @@ export interface v5Static {
     URL: string;
 }
 export type v5 = v5Buffer & v5String & v5Static;
+
+// https://github.com/uuidjs/uuid/blob/master/src/parse.js
+export type parse = (uuid: string) => Uint8Array;
```
Comment (ctavan) [inline types/uuid/interfaces.d.ts:53]: It's an implementation detail that we are currently returning a Uint8Array here, this should not be relied upon. Instead this should be an `OutputBuffer`.

File: types/uuid/uuid-tests.ts [hunk 2]
```diff
@@ -61,3 +69,9 @@ uuidv4(null, g); // $ExpectType Buffer
 class CustomBuffer extends Uint8Array {}
 const h = new CustomBuffer(10);
 uuidv4(null, h); // $ExpectType CustomBuffer
+
+const valid1: boolean = uuidValidate('6ec0bd7f-11c0-43da-975e-2a8ad9ebae0b');
+
+const parse1: Uint8Array = uuidParse('6ec0bd7f-11c0-43da-975e-2a8ad9ebae0b');
```
Comment (ctavan) [inline types/uuid/uuid-tests.ts:75]: See comment above.

**Review by ctavan — CHANGES_REQUESTED** [review]
I would prefer to wait for #46395 since the other pull request contains a more comprehensive update to the types that brings the types entirely up-to-date with the actual library.

**Comment by Crazyglue** [comment]
I'm going to close this PR in favor of #46395

Hope it gets merged soon, we need it at work :)

**Pull request closed without merging**